From 00505e1d059c76471fc612a7b4a17c60119f6e2d Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 3 Feb 2025 20:09:26 +0100
Subject: [PATCH] dt-bindings: soc: st: add stm32mp power domain

Add binding documentation for the power domains found on
STMicroelectronics STM32MP SoCs.

Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: If3d55e03486413ca40ac73f9bf07046e76a00f0a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/434113
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../bindings/soc/st/stm32mp-pm-domain.yaml    | 46 +++++++++++++++++++
 1 file changed, 46 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/soc/st/stm32mp-pm-domain.yaml

diff --git a/Documentation/devicetree/bindings/soc/st/stm32mp-pm-domain.yaml b/Documentation/devicetree/bindings/soc/st/stm32mp-pm-domain.yaml
new file mode 100644
index 000000000000..e0abc325db17
--- /dev/null
+++ b/Documentation/devicetree/bindings/soc/st/stm32mp-pm-domain.yaml
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/soc/st/stm32mp-pm-domain.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STM32MP generic power domain
+
+maintainers:
+  - Patrick Delaunay <patrick.delaunay@foss.st.com>
+
+allOf:
+  - $ref: /schemas/power/power-domain.yaml#
+
+description: |
+  Some STM32MP SoCs contain a D1 domain which defined the devices
+  assigned to Cortex A35, these device are functionnal only when the cortex the
+  A35 cluster is not in low power modes with clock OFF.
+
+  This PM domain node represents the D1 dependency depndency for clock PM
+  framework with the parent power domain but without wake-up path support.
+
+properties:
+  compatible:
+    const: st,stm32mp-pm-domain
+
+  '#power-domain-cells':
+    const: 0
+
+  power-domains:
+    maxItems: 1
+
+additionalProperties: false
+
+required:
+  - compatible
+  - '#power-domain-cells'
+  - power-domains
+
+examples:
+  - |
+    power-domain-d1 {
+      compatible = "st,stm32mp-pm-domain";
+      #power-domain-cells = <0>;
+      power-domains = <&CLUSTER_PD>;
+    };
-- 
2.50.1

